File: Demo/GtSdkDemo-Manual/GtSdkDemo/ApnsHelper.m
//
//  ApnsHelper.m
//  GtSdkDemo
//
//  Created by ak on 2020/4/14.
//  Copyright © 2020 Gexin Interactive (Beijing) Network Technology Co.,LTD. All rights reserved.
//

#import "ApnsHelper.h"

@implementation ApnsHelper

+(NSString*)makeMp3FromExt: (float)cnt {
    NSString *path = [[[[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.ent.com.getui.demo"] resourceSpecifier] stringByAppendingString:@"Library/Sounds/"];
    return [self mergeVoice:path cnt:cnt];
}

+(NSString*)mergeVoice:(NSString*)libPath cnt:(float)cnt {
    NSArray *nums = @[@"tip", @"tip"];
    NSMutableData *mergeData = [NSMutableData alloc];
    for (int i=0; i<nums.count; i++) {
        //TODO: 需要自行导入资源
//        NSString *mp3Url = [[NSBundle mainBundle] pathForResource:[NSString stringWithFormat:@"%@.mp3", nums[i]] ofType:@""];
        NSString *mp3Url = [[NSBundle mainBundle] pathForResource:@"tip" ofType:@"mp3"];
        NSData *data = [[NSData alloc] initWithContentsOfFile: mp3Url];
        [mergeData appendData:data];
    }
    if(mergeData.length == 0) {
        return @"";
    }
    if (![[NSFileManager defaultManager] fileExistsAtPath:libPath]) {
        NSError *error;
        [[NSFileManager defaultManager] createDirectoryAtPath:libPath withIntermediateDirectories:YES attributes:nil error:&error];
    }
    NSString *fileName = [NSString stringWithFormat:@"%d.mp3", [self getFileName]];
    NSString *filePath = [NSString stringWithFormat:@"%@%@", libPath, fileName];
    BOOL result = [mergeData writeToFile:filePath atomically:YES];
    NSLog(@"data length:%@ %@ result:%@ ", @([mergeData length]), filePath,@(result));
    return fileName;
}

+(int)getFileName {
    return arc4random()%100000;
}
@end

File: Demo/GtSdkDemo-swift/GtSdkDemo-swift/ApnsHelper.swift
//
//  ApnsHelper.swift
//  ApnsDemo
//
//  Created by ak on 2020/4/14.
//  Copyright © 2020 ak. All rights reserved.
//

import UIKit

struct ApnsHelper {
  static func makeMp3FromExt(_ cnt: Double) -> String {
    let path = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: "group.ent.com.getui.demo")!.absoluteString.replacingOccurrences(of: "file://", with: "") + "Library/Sounds/"
    return mergeVoice(libPath: path, cnt)
  }
  
  private static func mergeVoice(libPath: String, _ cnt: Double) -> String {
//    var nums = [String]()
//    var tmp = Int(cnt)
//    while(tmp>0) {
//      nums.insert("\(tmp % 10)", at: 0)
//      tmp = Int(tmp/10)
//    }
    //TODO: 需自行替换资源
    var mergeData = Data()
    ["tip", "tip"].forEach { num in
      let mp3Url = Bundle.main.url(forResource: "\(num).mp3", withExtension: "")
      if let mp3Url = mp3Url, let data = try? Data(contentsOf: mp3Url) {
        mergeData.append(data)
      }
    }
    guard !mergeData.isEmpty else { return "" }
    if !FileManager.default.fileExists(atPath: libPath) {
      do {
        try FileManager.default.createDirectory(atPath: libPath, withIntermediateDirectories: true, attributes: nil)
      } catch {
        NSLog("创建Sounds文件失败 \(libPath)")
      }
    }
    let fileName = "\(getFileName()).mp3"
    let fileUrl = URL(string: "file://\(libPath)\(fileName)")
    guard let url = fileUrl else { return "" }
    do {
      try mergeData.write(to: url)
    } catch {
      NSLog("合成mp3文件失败 \(url)")
    }
    return fileName
  }
    
  private static func getFileName() -> Int {
    return Int(arc4random()%100000)
  }
}
